Sources/Core/AWSClientRuntime/Sources/AWSClientRuntime/UserAgent/BusinessMetrics.swift

```diff
@@ -0,0 +1,106 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+import class Smithy.Context
+import struct Smithy.AttributeKey
+
+struct BusinessMetrics {
+    // Mapping of human readable feature ID to the corresponding metric value
+    let features: [String: String]
+
+    init(
+        config: UserAgentValuesFromConfig,
+        context: Context
+    ) {
+        setFlagsIntoContext(config: config, context: context)
+        self.features = context.businessMetrics
+    }
+}
+
+extension BusinessMetrics: CustomStringConvertible {
+    var description: String {
+        var commaSeparatedMetricValues = features.values.sorted().joined(separator: ",")
+        // Cut last metric value from string until the
+        //  comma-separated list of metric values are at or below 1024 bytes in size
+        if commaSeparatedMetricValues.lengthOfBytes(using: .ascii) > 1024 {
+            while commaSeparatedMetricValues.lengthOfBytes(using: .ascii) > 1024 {
+                commaSeparatedMetricValues = commaSeparatedMetricValues.substringBeforeLast(",")
+            }
+        }
+        return "m/\(commaSeparatedMetricValues)"
+    }
+}
+
+private extension String {
+    func substringBeforeLast(_ separator: String) -> String {
+        if let range = self.range(of: separator, options: .backwards) {
+            return String(self[..<range.lowerBound])
+        } else {
+            return self
+        }
+    }
+}
+
+public extension Context {
+    var businessMetrics: Dictionary<String, String> {
+        get { attributes.get(key: businessMetricsKey) ?? [:] }
+        set(newPair) {
+            var combined = businessMetrics
+            combined.merge(newPair) { (current, new) in new }
+            attributes.set(key: businessMetricsKey, value: combined)
```

Comment:
Adds a new entry to existing dictionary, overriding with new entry's value if existing dictionary already has an entry with the same key